LeaderboardBackend/Controllers/RunsController.cs

```diff
@@ -157,4 +158,51 @@ public async Task<ActionResult<CategoryViewModel>> GetCategoryForRun(Guid id)
 
         return Ok(CategoryViewModel.MapFrom(category));
     }
+
+    // TODO: Replace UserTypes with UserRole, i.e. reconfigure authZ policy infra
+    [Authorize]
+    [HttpPatch("run/{id}")]
+    [SwaggerOperation(
+        "Updates a run with the specified new fields. This request is restricted to administrators. " +
+        "Note: `runType` cannot be updated. " +
+        "This operation is atomic; if an error occurs, the run will not be updated. " +
+        "All fields of the request body are optional but you must specify at least one.",
+        OperationId = "updateRun"
+    )]
+    [SwaggerResponse(204)]
+    [SwaggerResponse(401)]
+    [SwaggerResponse(403)]
+    [SwaggerResponse(404, Type = typeof(ProblemDetails))]
+    [SwaggerResponse(422, Type = typeof(ValidationProblemDetails))]
+    public async Task<ActionResult> UpdateRun(
+        [FromRoute] Guid id,
+        [FromBody, SwaggerRequestBody(Required = true)] UpdateRunRequest request
+    )
+    {
+        GetUserResult userRes = await userService.GetUserFromClaims(HttpContext.User);
+
+        if (!userRes.IsT0)
+        {
+            return Unauthorized();
+        }
+
+        UpdateRunResult res = await runService.UpdateRun(userRes.AsT0, id, request);
+
+        return res.Match<ActionResult>(
+            badRole => Forbid(),
+            notFound => NotFound(
+                ProblemDetailsFactory.CreateProblemDetails(
+                    HttpContext,
+                    404,
+                    "Run Not Found"
+                    )
+                ),
+            badRunType =>
+            {
+                ModelState.AddModelError("runType", "Bad run type");
```

Comment:
This is more of a semantic error than an actual model binding error so I'm not sure that it makes sense to do this here.

Reply:
True, but this is more for the response object it gives, which still makes sense IMO.

Reply:
You can return a problem details with a 422 and a custom title. I think that would be the preferred way to do it.

Reply:
In that block I have a TODO above it that says that this block should be returning a ValidationProblemDetails instead, actually. But now I'm wondering; do we expect 422s to always return a ValidationProblemDetails?

Reply:
It honestly doesn't need to as long as it's documented properly. It should be consistent with the run creation route in returning a problem details.